lib/collibra/client/get_rights_to_use_asset.ts

```diff
@@ -1,29 +1,39 @@
-import { RequesterFn } from "./make_collibra_client"
+import * as TE from "fp-ts/TaskEither"
+import * as E from "fp-ts/Either"
+import { pipe } from "fp-ts/lib/function"
+import { AxiosResponse } from "axios"
 
-export const getRightsToUseAsset: RequesterFn<Collibra.Asset> = (client) => async (id: string) => {
-  const {
-    data: {
-      results: [relationType],
-    },
-  } = await client.get<Collibra.PagedRelationTypeResponse>(`/relationTypes`, {
-    params: {
-      sourceTypeName: "data product",
-      targetTypeName: "rights-to-use",
-    },
-  })
+type RelationTypesResponse = AxiosResponse<Collibra.PagedRelationTypeResponse>
+type RelationsResponse = AxiosResponse<Collibra.PagedRelationResponse>
+type AssetResponse = AxiosResponse<Collibra.Asset>
 
-  const {
-    data: {
-      results: [relation],
-    },
-  } = await client.get<Collibra.PagedRelationResponse>(`/relations`, {
-    params: {
-      relationTypeId: relationType.id,
-      sourceId: id,
-    },
-  })
-
-  const { data: asset } = await client.get<Collibra.Asset>(`/assets/${relation.target.id}`)
-
-  return asset
-}
+export const getRightsToUseAsset = (client: Net.Client) => (id: string) =>
+  pipe(
+    TE.tryCatch(
+      () =>
+        client.get<RelationTypesResponse>("/relationTypes", {
+          params: new URLSearchParams({
+            sourceTypeName: "data product",
+            targetTypeName: "rights-to-use",
+          }),
+        }),
+      E.toError
+    ),
+    TE.bindTo("relationTypesResponse"),
```

Comment:
Bind statements probably not needed. See earlier comments.

Reply:
`bindTo` on line 22 might not be needed, but `TE.tryCatch` doesn't accept an argument in its callback function. would `TE.map` be preferred?

Reply:
yep that did it! 😄